### PR TITLE
Fix: Script waits forever in the Arena of Heroes #33

### PR DESCRIPTION
**Pull request by kevingrillet**

### CODE REVIEW

**Comment by Zebiano** [comment]
I'm not a big fan of solving this problem this way... I'd rather have a check on each `arenaHeroesOpponent` `case` if the button exists. If it doesn't, a function would be called to try and tap the closest button. Something like:
```sh
# Attempts to tap the closest Arena of Heroes opponent. Params: opponent
function tapClosestOpponent() {
  # Depending on the opponent number sent as a parameter ($1), this function 
  # would attempt to check if there's an opponent above the one sent.
  # If there isn't, check the one above that one and so on until one is found. 
  # When found, tap on the opponent and exit function.
}

case $arenaHeroesOpponent in
  1)
    # Check if opponent exists
    # If yes, input tap 820 700
    # If no, tapClosestOpponent 1
  ;;
  2)
    # Check if opponnent exists
    # If yes, input tap 820 870
    # If no, tapClosestOpponent 2
  ;;
  # And so on...
esac
```

**Comment by kevingrillet** [comment]
Ok, I'll try to do it when I have some time :p

**Comment by kevingrillet** [comment]
*Done I think*

**Comment by Zebiano** [comment]
Looks good to me, I'll have to test it out tomorrow though. Thank you!

**Comment by kevingrillet** [comment]
I did mix up things a little, it's a PR for:

- Fix: Script waits forever in the Arena of Heroes #33
- Fix: 2 settings for Arena and Legends tries #47
- Fix: Strenghen Crystal break on level up 👎 #50

**Comment by Zebiano** [comment]
What happens when there are no opponents? I'm guessing that could happen in case you are first on the leaderboard. And my guess is that you wouldn't be able to fight anyone, which the script should realise and then skip battling in the Arena. I'm guessing with the current code, the script would get into an endless loop in case there are no enemies to fight, which is not ideal.

File: afk-daily.sh [hunk 1]
```diff
@@ -744,6 +744,55 @@ function teamBounties() {
...
+            getColor 820 700 # Opponent 1
+            if [ "$RGB" == "a7f1b7" ]; then # Looking for the button
+                input tap 820 700
+            else
+                tapClosestOpponent 1
+            fi
+            ;;
+        3)
+            getColor 820 870 # Opponent 2
+            if [ "$RGB" == "2daab4" ] || [ "$RGB" == "aff3c0" ]; then # Looking for the button
```
Comment (Zebiano) [inline afk-daily.sh:771]: Why do we check for two values here and also on line `787`?

Comment (Zebiano) [inline afk-daily.sh:771]: Why do we check for two values here and also on line `787`?

Comment (kevingrillet) [reply afk-daily.sh:771]: > Why do we check for two values here and also on line `787`?

The value change if you are fighting for free or using arena tickets.

```shell
            #                                Free         x1
            #  Opponent 1: 820 700      ->        a7f1b7
            #  Opponent 2: 820 870      ->  2eaab4      aff3c0
            #  Opponent 3: 820 1050     ->        a7f1b7
            #  Opponent 4: 820 1220     ->  2daab4      aff3c0
            #  Opponent 5: 820 1400     ->        aaf2bb
```

**Comment by kevingrillet** [comment]
> What happens when there are no opponents? I'm guessing that could happen in case you are first on the leaderboard. And my guess is that you wouldn't be able to fight anyone, which the script should realise and then skip battling in the Arena. I'm guessing with the current code, the script would get into an endless loop in case there are no enemies to fight, which is not ideal.

Maybe adding a global var on the loop to break after `x` fails.

**Comment by kevingrillet** [comment]
I'm moving #47 and #50 on dedicated branches

**Comment by Zebiano** [comment]
Looks good to me, will test it out later.

**Comment by Zebiano** [comment]
Thank you for this PR! 🎉

**Comment by Zebiano** [comment]
Fix: #33

**Pull request merged**